contracts/AelinPool.sol

```diff
@@ -93,17 +97,19 @@ contract AelinPool is AelinERC20, MinimalProxyFactory {
         aelinDealLogicAddress = _aelinDealLogicAddress;
         aelinRewardsAddress = _aelinRewardsAddress;
 
-        emit SetSponsor(_sponsor);
-    }
-
-    function updateAllowList(address[] memory _allowList, uint256[] memory _allowListAmounts) external onlyPoolFactoryOnce {
-        for (uint256 i = 0; i < _allowList.length; i++) {
-            allowList[_allowList[i]] = _allowListAmounts[i];
-            emit AllowlistAddress(
-                _allowList[i],
-                _allowListAmounts[i]
+        if (_poolData.allowListAddresses.length > 0 || _poolData.allowListAmounts.length > 0) {
```

Comment:
we could use an array of structs instead of 2 arrays. Something like struct AllowList with address and amount